src/ui/ui_settings/ui_settings_service.js

```diff
@@ -95,6 +107,20 @@ export class UiSettingsService {
     await this.setMany(changes);
   }
 
+  async _validate(changes) {
+    const defaults = await this.getDefaults();
+    for (const key of Object.keys(changes)) {
+      const validations = defaults[key].validations;
+      const value = changes[key];
+      for (const validation of Object.values(validations)) {
+        const result = validation.schema.validate(value);
```

Comment:
I think it's important to capture `result.value` here, since `JoiSchema#validate()` will return defaults and such if they are configured. Also, why not have a single joi schema for the value?